In [1]:
import numpy as np

In [2]:
PATH_FILE = '../../../Assets/playstore/3 - Data Used/googleplaystore_used_numpy.csv'

In [3]:
init_data = {
    'delimiter': ',',
    'skip_header': 1
}

# 1 - Charger le fichier de données

In [4]:
google = np.genfromtxt(PATH_FILE, **init_data)

# 2 - Enquêter sur son contenu

In [5]:
print('Sa taille : {}\nSon nombre de dimension : {}\nLe nombre total d\'informations : {}'.format(google.shape, google.ndim, google.size))

Sa taille : (7729, 5)
Son nombre de dimension : 2
Le nombre total d'informations : 38645


Nous avons affaire à une matrice de données. Elle contient exactement 38645 informations. Ces dernières sont réparties en 5 colonnes. Chaque colonne contient 7729 informations. Soit :

In [6]:
7729 * 5

38645

# 3 - Sélectionner les 10 premières lignes

In [7]:
google[:10,:]

array([[4.10000e+00, 1.59000e+02, 1.90000e+01, 1.00000e+04, 0.00000e+00],
       [3.90000e+00, 9.67000e+02, 1.40000e+01, 5.00000e+05, 0.00000e+00],
       [4.70000e+00, 8.75100e+04, 8.70000e+00, 5.00000e+06, 0.00000e+00],
       [4.50000e+00, 2.15644e+05, 2.50000e+01, 5.00000e+07, 0.00000e+00],
       [4.30000e+00, 9.67000e+02, 2.80000e+00, 1.00000e+05, 0.00000e+00],
       [4.40000e+00, 1.67000e+02, 5.60000e+00, 5.00000e+04, 0.00000e+00],
       [3.80000e+00, 1.78000e+02, 1.90000e+01, 5.00000e+04, 0.00000e+00],
       [4.10000e+00, 3.68150e+04, 2.90000e+01, 1.00000e+06, 0.00000e+00],
       [4.40000e+00, 1.37910e+04, 3.30000e+01, 1.00000e+06, 0.00000e+00],
       [4.70000e+00, 1.21000e+02, 3.10000e+00, 1.00000e+04, 0.00000e+00]])

# 4 - Sélectionner les 10 dernière lignes

In [8]:
google[-10:,:]

array([[4.20000e+00, 1.17000e+02, 1.30000e+01, 5.00000e+03, 0.00000e+00],
       [3.40000e+00, 2.91000e+02, 1.30000e+01, 1.00000e+04, 0.00000e+00],
       [4.60000e+00, 6.03000e+02, 7.40000e+00, 1.00000e+04, 0.00000e+00],
       [3.80000e+00, 8.81000e+02, 2.30000e+00, 1.00000e+05, 0.00000e+00],
       [3.80000e+00, 1.19500e+03, 5.82000e-01, 1.00000e+05, 0.00000e+00],
       [4.80000e+00, 4.40000e+01, 6.19000e-01, 1.00000e+03, 0.00000e+00],
       [4.00000e+00, 7.00000e+00, 2.60000e+00, 5.00000e+02, 0.00000e+00],
       [4.50000e+00, 3.80000e+01, 5.30000e+01, 5.00000e+03, 0.00000e+00],
       [5.00000e+00, 4.00000e+00, 3.60000e+00, 1.00000e+02, 0.00000e+00],
       [4.50000e+00, 3.98307e+05, 1.90000e+01, 1.00000e+07, 0.00000e+00]])

# 5 - Sélectionner la première et la dernière colonne de deux manières différentes

In [9]:
# nous sélectionnons via les indices
first_last_column_one = [0, 4]
# on emploie cette fois-ci un indice négatif
first_last_column_two = [0, -1]

In [10]:
google[:,first_last_column_one]

array([[4.1, 0. ],
       [3.9, 0. ],
       [4.7, 0. ],
       ...,
       [4.5, 0. ],
       [5. , 0. ],
       [4.5, 0. ]])

In [11]:
google[:,first_last_column_two]

array([[4.1, 0. ],
       [3.9, 0. ],
       [4.7, 0. ],
       ...,
       [4.5, 0. ],
       [5. , 0. ],
       [4.5, 0. ]])

# 6 - Trouver le maximum de chaque colonne.

In [12]:
max_each_col = google.max(axis = 0)
print('Max rating :', max_each_col[0])
print('Max review :', max_each_col[1])
print('Max size :', max_each_col[2])
print('Max install :', max_each_col[3])
print('Max price :', max_each_col[4])

Max rating : 5.0
Max review : 44893888.0
Max size : 100.0
Max install : 1000000000.0
Max price : 400.0


# 7 - Trier le tableau par le nombre de « reviews » (par ordre décroissant) et sauvegarder le résultat dans « google_sorted_reviews ».

In [13]:
# Par défaut, argsort exécute le tri par ordre croissant. 
# Ici, on inverse les résultats et on transforme le tout en liste
index_sort_reviews = list(reversed(np.argsort(google[:, 1])))

In [14]:
google_sorted_reviews = google[index_sort_reviews].copy()

In [15]:
google_sorted_reviews[:100,1]

array([44893888., 44891723., 44891723., 44881447., 27725352., 27725352.,
       27724094., 27723193., 27722264., 27711703., 23136735., 23134775.,
       23133508., 23125280., 22430188., 22430188., 22429716., 22428456.,
       22428456., 22426677., 22419455., 17714850., 17712922., 17712922.,
       14201891., 14201604., 14200550., 14200344., 14198602., 14198297.,
       14184910., 13479633., 12759815., 12759663., 10981850., 10979062.,
       10486018., 10485334., 10485308., 10483141., 10424925., 10421284.,
       10055521., 10053186.,  9883806.,  9883367.,  9882639.,  9881908.,
        9881829.,  9879473.,  9876369.,  9873470.,  8923847.,  8923818.,
        8923587.,  8921451.,  8389714.,  8219586.,  8190074.,  8119154.,
        8119151.,  8118937.,  8118937.,  8118609.,  8116142.,  7790693.,
        7775146.,  7615646.,  7614415.,  7614407.,  7614271.,  7614130.,
        7594559.,  7590099.,  6702776.,  6700847.,  6427773.,  6212081.,
        6211039.,  6211039.,  6210998.,  6200739., 

# 8 - Réaliser une moyenne sur les 100 premières lignes de vos données précédemment triées. Interpréter les résultats.

In [16]:
# Nous spécifions l'axe 0 (soit vertical).
# En résumé, il fait la moyenne de toutes les lignes de chaque colonne.
# Pour plus d'informations à ce sujet, consultez ce lien : https://docs.scipy.org/doc/numpy-1.10.0/glossary.html
mean_top_100_app = google_sorted_reviews[:100].mean(axis=0)

In [17]:
print('Mean rating by top 100 most reviews :', mean_top_100_app[0])
print('Mean review by top 100 most reviews :', mean_top_100_app[1])
print('Mean size by top 100 most reviews :', mean_top_100_app[2])
print('Mean install by top 100 most reviews :', mean_top_100_app[3])
print('Mean price by top 100 most reviews :', mean_top_100_app[4])

Mean rating by top 100 most reviews : 4.463999999999999
Mean review by top 100 most reviews : 13040739.36
Mean size by top 100 most reviews : 64.3
Mean install by top 100 most reviews : 241500000.0
Mean price by top 100 most reviews : 0.0


# 9 - Obtenir le nombre d’applications qui ont un rating plus grand que la moyenne (provenant de votre variable google). Calculer le pourcentage qui se trouvent au-dessus et en dessous de la moyenne.

In [18]:
mean_rating = google[:,0].mean()
weighted_mean_rating = (google[:,0] * google[:,1]).sum() / google[:,1].sum()
print('Mean : {}\nWeighted mean : {}'.format(mean_rating, weighted_mean_rating))

Mean : 4.173851727260965
Weighted mean : 4.448669318080345


In [19]:
condition_above_mean_rating = google[:,0] >= mean_rating
condition_above_w_mean_rating = google[:,0] >= weighted_mean_rating

In [20]:
nb_app_above_mean_rating = google[condition_above_mean_rating].shape[0]
nb_app_above_w_mean_rating = google[condition_above_w_mean_rating].shape[0]
print('There are {} applications above the mean rating'.format(nb_app_above_mean_rating))
print('There are {} applications above the w mean rating'.format(nb_app_above_w_mean_rating))

There are 4853 applications above the mean rating
There are 2460 applications above the w mean rating


In [21]:
print((nb_app_above_mean_rating / google.shape[0]) * 100, '%')
print((nb_app_above_w_mean_rating / google.shape[0]) * 100, '%')

62.7894941130806 %
31.828179583387247 %


# 10 - « Il y a plus d’applications gratuites que d’applications payantes ». Est-ce vrai ?

In [22]:
value_count_prices = {}

In [23]:
for p in google[:,-1]:
    value = value_count_prices.get(p, 0)
    value_count_prices[p] = value + 1

In [24]:
value_count_prices

{0.0: 7150,
 4.99: 63,
 6.99: 9,
 7.99: 7,
 3.99: 45,
 5.99: 15,
 2.99: 101,
 1.99: 53,
 9.99: 16,
 0.99: 106,
 9.0: 2,
 5.49: 2,
 10.0: 3,
 24.99: 5,
 11.99: 4,
 79.99: 2,
 16.99: 3,
 14.99: 10,
 29.99: 6,
 12.99: 4,
 3.49: 6,
 10.99: 2,
 7.49: 1,
 1.5: 1,
 19.99: 5,
 15.99: 1,
 33.99: 2,
 39.99: 1,
 2.49: 17,
 4.49: 8,
 1.7: 2,
 1.49: 28,
 3.88: 1,
 399.99: 11,
 17.99: 2,
 400.0: 1,
 3.02: 1,
 1.76: 1,
 4.84: 1,
 4.77: 1,
 1.61: 1,
 1.59: 1,
 6.49: 1,
 1.29: 1,
 299.99: 1,
 379.99: 1,
 37.99: 1,
 18.99: 1,
 389.99: 1,
 8.49: 1,
 1.75: 1,
 14.0: 1,
 2.0: 1,
 3.08: 1,
 2.59: 1,
 19.4: 1,
 15.46: 1,
 8.99: 2,
 3.04: 1,
 13.99: 1,
 4.29: 1,
 3.28: 1,
 4.6: 1,
 1.0: 2,
 2.9: 1,
 1.97: 1,
 2.56: 1,
 1.2: 1}

In [25]:
count_app_free = value_count_prices[.0]

In [26]:
nb_total_app = sum(list(value_count_prices.values()))

In [27]:
nb_total_app

7729

In [28]:
# On check si on a bien toutes nos données
nb_total_app == google.shape[0]

True

In [29]:
# On calcule ensuite le pourcentage
percent_free_app = (count_app_free / nb_total_app) * 100
percent_free_app

92.50873334195886

In [30]:
print('There are {} % of free applications in this dataset'.format(percent_free_app))
print('Yes, there are more free app than paid app')

There are 92.50873334195886 % of free applications in this dataset
Yes, there are more free app than paid app


# 11 - « Les applications gratuites reçoivent plus d’avis que les applications payantes ». Est-ce vrai ?

In [31]:
condition_app_free = google[:, -1] == .0

In [32]:
google_app_free = google[condition_app_free].copy()
google_app_paid = google[~condition_app_free].copy()

In [33]:
total_reviews_app_free = google_app_free[:, 1].sum()
total_reviews_app_paid = google_app_paid[:, 1].sum()

In [34]:
print(total_reviews_app_free)
print(total_reviews_app_paid)

2273461569.0
4062575.0


In [35]:
print('Is free app receive more reviews than paid app ? :', total_reviews_app_free > total_reviews_app_paid)

Is free app receive more reviews than paid app ? : True
